NitroxServer/GameLogic/Bases/BaseManager.cs

```diff
@@ -33,29 +35,83 @@ public List<BasePiece> GetPartiallyConstructedPieces()
                 return new List<BasePiece>(partiallyConstructedPiecesById.Values);
             }
         }
-
         public void AddBasePiece(BasePiece basePiece)
         {
-            lock (partiallyConstructedPiecesById)
+            //anything that is not a window, hatch, or reinforcement get run normally
+            if (basePiece.TechType.Name != "BaseReinforcement" | basePiece.TechType.Name != "BaseHatch" | basePiece.TechType.Name != "BaseWindow")
+            {
+                //just in case some how a base piece is spawned at max completion
+                //the base piece will get added to at least one directory
+                if(basePiece.ConstructionAmount < 1.0f)
```

Comment:
Can we validate that this is the behavior in creative?  (along with the types of game objects that have this behavior)  We would also need to validate that these items also don't send construction complete events.  However, it could be a valid edge case given these two things check out.

Reply:
Yes when build in creative and speed is above 1 or the fastbuild cheat is active it goes from 0% to 100%